gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2271,6 +2298,69 @@ private boolean canIgnoreExceptionForEmptyObject(
     return false;
   }
 
+  @Override
+  public boolean isHnBucket(URI src) throws IOException {
+    String bucketName = src.getAuthority();
+    Boolean isEnabled = cache.getIfPresent(bucketName);
+    if (isEnabled != null) {
+      return isEnabled;
+    }
+
+    String prefix = src.getPath().substring(1);
+
+    StorageControlClient storageControlClient = lazyGetStorageControlClient();
```

Comment:
@arunkumarchacko can you move initialization of Storage Control into constructor of GoogleCloudStorageImpl? This will go along Apiary initialization on line 398. The high latency comes from creating the client. I'm assuming this would only need to be created if HNS is enabled for the hadoop workload (iiuc the flag correctly).

Reply:
Also wondering on benchmarks; Do you have comparison between existing rename without HNS and HNS enabled latencies?

Reply:
@frankyn We need this client if all the following are true:

1. The flag is enabled
2. The bucket is an HNS bucket
3. rename operation will be called

[3] above will be called fewer number of times. Hence not sure if incurring the cost all the time makes sense.

Reply:
> Also wondering on benchmarks; Do you have comparison between existing rename without HNS and HNS enabled latencies?

I am working to have this run daily. We should have this some time next week.

Reply:
Thanks Arun, once we have comparisons that will help with a goal post.

For storage control client initialization, taking #1, I'm wondering if client initialization could be done when the flag is enabled because a customer has explicitly requested it?